test/components/form-draft/testing.spec.js

```diff
@@ -7,9 +8,27 @@ import { load } from '../../util/http';
 import { mockLogin } from '../../util/session';
 
 describe('FormDraftTesting', () => {
-  beforeEach(mockLogin);
+  describe('new submission button', () => {
+    it('shows the button to an administrator', async () => {
+      mockLogin({ role: 'admin' });
+      testData.extendedForms.createPast(1, { draft: true });
+      const path = '/projects/1/forms/f/draft/testing';
+      const component = await load(path, { component: true }, {});
+      component.first(EnketoFill).should.be.visible();
+    });
+
+    it('does not render the button for a project viewer', async () => {
```

Comment:
I'm kind of surprised that a project viewer can even see drafts? I guess they could be verifying test submissions that come in, is that the idea?

Reply:
Yes, I think so. In addition to the Submissions and Testing tabs, project viewers can also see the Versions tab. It's not clear that project viewers will use the Versions and Testing tabs, but they have all the verbs necessary to view that data, and it's ultimately less frontend logic to let them see them. In the past, we initially locked down archived projects, then ended up reversing that, and since then, @issa-tseng and I have tended toward keeping things accessible unless there's a reason not to. But it also isn't hard to hide the tabs if that'd be better.

Reply:
Pragmatic approach. 👍 Thanks for the explanation. I’m on board.